application/tests/noise_filter/sanitize_test.py

```
@@ -0,0 +1,163 @@
+    def test_strips_null_bytes(self) -> None:
+        self.assertEqual(sanitize_text("hello\x00world"), "hello world")
+
+    def test_nfc_normalizes_decomposed(self) -> None:
+        # "é" can be a single codepoint U+00E9 (NFC) or e + combining acute (NFD).
+        decomposed = "café"  # e + combining acute -> "café"
+        composed = "café"
+        self.assertEqual(sanitize_text(decomposed), composed)
+
+    def test_strips_zero_width_characters(self) -> None:
+        # U+200B ZWSP between every char of "hello".
+        text = "h​e​l​l​o"
```

Ruff (0.15.15)

[error] Invalid unescaped character zero-width-space, use "\u200B" instead

Replace with escape sequence

(PLE2515)